### PR TITLE
fix(react): input amount formatting and max discounting fee

### PR DESCRIPTION
**Pull request by pedronauck**

### CODE REVIEW

File: design-system/react/src/components/InputAmount/InputAmount.tsx [hunk 1]
```diff
@@ -32,19 +35,22 @@ type InputAmountComponent = FC<InputAmountProps> & {
 export const InputAmount: InputAmountComponent = ({
   name,
   balance,
+  balanceUnits = 3,
```
Comment (LuizAsFight) [inline design-system/react/src/components/InputAmount/InputAmount.tsx:38]: should it be `balancePrecision` ? units and precision are different.

File: design-system/react/src/components/InputAmount/utils.tsx [hunk 2]
```diff
@@ -22,3 +23,8 @@ export function createAmount(text: string, units: number = DECIMAL_UNITS) {
     amount: bn.parseUnits(text, units),
   };
 }
+
+export function formatAmount(amount: BNInput, opts: FormatConfig = {}) {
+  const { precision = DECIMAL_UNITS, units = DECIMAL_UNITS } = opts;
```
Comment (LuizAsFight) [inline design-system/react/src/components/InputAmount/utils.tsx:28]: although they are the same number right now, `units` and `precision` are different.

we should use: `DECIMAL_PRECISION` and `DECIMAL_UNITS`

**Review by LuizAsFight — CHANGES_REQUESTED** [review]

File: design-system/react/src/components/InputAmount/InputAmount.tsx (hunk 1 above)
Comment (pedronauck) [reply design-system/react/src/components/InputAmount/InputAmount.tsx:38]: you're right

**Review by LuizAsFight — APPROVED** [review]

**Review by LuizAsFight — CHANGES_REQUESTED** [review]
code is ok, but pr has broken CI

**Review by luizstacio — CHANGES_REQUESTED** [review]
The balance of the user fee should be informed on the application level. In this way, the user knows the max he can apply. Making it only when clicking on max makes this a hidden action for the user, that will not know what is happening.

Also the balance shown on the input should show the entire number of decimals, or we can add a tooltip when the user hovers to show the total balance.

File: design-system/react/src/components/InputAmount/InputAmount.tsx [hunk 3]
```diff
@@ -1,28 +1,31 @@
...
+  balancePrecision?: number;
   value?: BN | null;
   units?: number;
   onChange?: (val: BN) => void;
   hiddenMaxButton?: boolean;
   hiddenBalance?: boolean;
-  max?: BN;
   inputProps?: InputNumberProps;
+  isDisabled?: boolean;
+  fee?: BN | null;
```
Comment (luizstacio) [inline design-system/react/src/components/InputAmount/InputAmount.tsx:28]: The balance of the user fee should be informed on the application level. In this way, the user knows the max he can apply. Making it only when clicking on max makes this a hidden action for the user, that will not know what is happening.

Also the balance shown on the input should show the entire number of decimals, or we can add a tooltip when the user hovers to show the total balance.

**Review by luizstacio — CHANGES_REQUESTED** [review]

File: design-system/react/src/components/InputAmount/InputAmount.tsx (hunk 3 above)
Comment (pedronauck) [reply design-system/react/src/components/InputAmount/InputAmount.tsx]: ✅

**Review by luizstacio — APPROVED** [review]

**Pull request merged**